src/SprykerEco/Zed/Braintree/Business/Payment/Transaction/Handler/RefundItemsTransactionHandler.php

```diff
@@ -0,0 +1,218 @@
+
+        foreach ($orderItemsGroupedByTransaction as $transactionId => $data) {
+            $transactionMetaTransfer->setIdPayment($data[static::KEY_PAYMENT_ID]);
+            $transactionMetaTransfer->setTransactionIdentifier($transactionId);
+            $transactionMetaTransfer->setRefundAmount($data[static::KEY_AMOUNT]);
+
+            $braintreeTransactionResponseTransfer = $this->transaction->executeTransaction($transactionMetaTransfer);
+
+            if ($braintreeTransactionResponseTransfer->getIsSuccess()) {
+                $refundTransfer = $this->removeShipmentExpense($refundTransfer);
+            }
+        }
+
+        if ($shipmentExpense) {
+            $refundTransfer->getExpenses()->append($shipmentExpense);
+        }
+
+        $this->refundFacade->saveRefund($refundTransfer);
+    }
+
+    /**
+     * @param \Orm\Zed\Sales\Persistence\SpySalesOrderItem[] $salesOrderItems
+     * @param \Orm\Zed\Sales\Persistence\SpySalesOrder $salesOrderEntity
+     *
+     * @return \Generated\Shared\Transfer\RefundTransfer
+     */
+    protected function getRefund(array $salesOrderItems, SpySalesOrder $salesOrderEntity): RefundTransfer
+    {
+        return $this->refundFacade->calculateRefund($salesOrderItems, $salesOrderEntity);
+    }
+
+    /**
+     * @param \Orm\Zed\Sales\Persistence\SpySalesOrderItem[] $orderItems
+     *
+     * @return \ArrayObject
+     */
+    protected function getItemsForRefunding(array $orderItems): ArrayObject
+    {
+        $itemsForRefunding = [];
+
+        foreach ($orderItems as $orderItem) {
+            $itemTransfer = new ItemTransfer();
+            $itemTransfer->fromArray($orderItem->toArray(), true);
+            $itemsForRefunding[] = $itemTransfer;
+        }
+
+        return $this->getRefundedUniqItems($itemsForRefunding);
+    }
+
+    /**
+     * @param \Orm\Zed\Sales\Persistence\SpySalesOrderItem[] $salesOrderItems
+     * @param \Orm\Zed\Sales\Persistence\SpySalesOrder $salesOrderEntity
+     * @param \Generated\Shared\Transfer\RefundTransfer $refundTransfer
+     *
+     * @return \Generated\Shared\Transfer\TransactionMetaTransfer
+     */
+    protected function getTransactionMetaTransfer(array $salesOrderItems, SpySalesOrder $salesOrderEntity, RefundTransfer $refundTransfer): TransactionMetaTransfer
+    {
+        $transactionMetaTransfer = new TransactionMetaTransfer();
+        $transactionMetaTransfer->setIdSalesOrder($salesOrderEntity->getIdSalesOrder());
+        $transactionMetaTransfer->setRefund($refundTransfer);
+        $transactionMetaTransfer->setItems($this->getItemsForRefunding($salesOrderItems));
+
+        return $transactionMetaTransfer;
+    }
+
+    /**
+     * @param \ArrayObject|\Generated\Shared\Transfer\ItemTransfer[] $itemTransfers
+     *
+     * @return \ArrayObject|\Generated\Shared\Transfer\ItemTransfer[]
+     */
+    protected function getRefundedUniqItems(array $itemTransfers): iterable
```

Comment:
You return exactly ArrayObject. I don't see why you need iterable as a return type here